include/boost/multiprecision/detail/et_ops.hpp

```diff
@@ -621,7 +803,18 @@ operator*(const number<B, ET>& a, const detail::expression<tag, Arg1, Arg2, Arg3
 }
 
 template <class B, expression_template_option ET, class tag, class Arg1, class Arg2, class Arg3, class Arg4>
-inline BOOST_MP_CXX14_CONSTEXPR typename detail::expression<detail::multiplies, number<B, ET>, detail::expression<tag, Arg1, Arg2, Arg3, Arg4> >::result_type
+inline BOOST_MP_CXX14_CONSTEXPR typename std::enable_if<
+   std::is_same<typename detail::expression<detail::multiplies, number<B, ET>, detail::expression<tag, Arg1, Arg2, Arg3, Arg4> >::result_type, number<B, ET>>::value,
+   typename detail::expression<detail::multiplies, number<B, ET>, detail::expression<tag, Arg1, Arg2, Arg3, Arg4> >::result_type>::type
+operator*(number<B, ET>&& a, const detail::expression<tag, Arg1, Arg2, Arg3, Arg4>& b)
+{
+   a *= b;
+   return static_cast<number<B, et_on>&&>(a);
```

Comment:
Same here: why `et_on` and not `ET`?

Reply:
Good catch: too much blind refactoring in one go!

I'm a bit concerned that didn't show up in local testing, also thinking I should be using `std::foward` here.

Reply:
`std::forward` and the `static_cast` (where right) are completely equivalent. The advantage of using `std::forward` is a better understanding of the indent of the developer.

Reply:
Brilliant! Now that I see your commit e4827ced2298503178f1d855979d605626eda9ad... of course it was not `std::forward`, but `std::move`, and now the code is a lot clearer to read! Thanks.